src/core/core.scale.js

```diff
@@ -60,6 +60,111 @@ module.exports = function(Chart) {
+	 * @param point2 numeric value
+	 * @return the mid point between point1 and point2
+	 */
+	function midPoint(point1, point2) {
+		return (point1 + point2) / 2;
+		// return point1 + ((point2 - point1) / 2);
+	}
+
+	/**
+	 * Calculate text origin and alignment
+	 *
+	 * Center origin and alignment prefered, but shift to start or end when using
+	 * center would cause truncation.
+	 *
+	 * @param isForward true when the text is to be placed right side up
+	 * @param centerPoint coordinate value to center text in the target area
+	 * @param halfWidth one half the width of the text to be placed
+	 * @param maxExtend the maximum coordinate beyond which truncation will occur
+	 * @return object with text placment origin coordinate and alignment
+	 */
+	function useEndOnTruncate(isForward, centerPoint, halfWidth, maxExtent) {
+		var positionProperties = {
+			coord: centerPoint,
+			alignment: 'center'
+		};
+		// Do the start trunction check first to get 'start' alignment when both
+		// ends would truncate
+		if (isForward) {// right side up text
+			if (centerPoint < halfWidth) {// would truncate the start
+				positionProperties.coord = 0;
+				positionProperties.alignment = 'start';
+			} else if (centerPoint + halfWidth > maxExtent) {// would truncate the end
+				positionProperties.coord = maxExtent;
+				positionProperties.alignment = 'end';
+			}// end of (forward) text truncation checks
+		} else if (centerPoint + halfWidth > maxExtent) {// would truncate the start
+			positionProperties.coord = maxExtent;
+			positionProperties.alignment = 'start';
+		} else if (centerPoint < halfWidth) {// would truncate the end
+			positionProperties.coord = 0;
+			positionProperties.alignment = 'end';
+		}// end of (reverse) text truncation checks
+
+		return positionProperties;
+	}// ./useEndOnTruncate(…){
+
+	/**
+	 * Place text, centered if it will fit
+	 *
+	 * Center text, if it fully fits in an area plus margins, otherwise set
+	 *	origin to the outer margin edge, and make that the start/end of the text
+	 *
+	 * Font must be set for the context before calling.	Fails to set (at least)
+	 *	the size if passed in the constraints, and set here.
+	 * Handles horizontal or vertical placment, but not other angles
+	 *
+	 * @param ctx CanvasRenderingContext2D drawing context
+	 * @param text single text string
+	 * @param cnstr object with text placement contraints
+	 * @return object with adjusted text placment properties
+	 */
+	function constrainedTextCenter(ctx, text, cnstr) {
+		var txtProperties;
+		var isForward = cnstr.position !== 'left'; // All text is right side up,
+			// except for Y Axis placed on the left of the chart
+		var halfTextWidth = ctx.measureText(text).width / 2;
+		var out = {
+			positionX: 0,
+			positionY: 0,
+			rotation: 0,
+			hAlign: 'center'
+		};
+
+		if (cnstr.isHorizontal) {// is horizontal text
+			txtProperties = useEndOnTruncate(isForward,
+				midPoint(cnstr.container.left, cnstr.container.right),
```

Comment:
I did not realize that caching would help the minifier.

I wrote the midPoint function, because I find `midPoint(start, end)` to be a lot more semantic than `start + ((end - start) / 2)` that the code I started from used.  Generally, code should be written to help the reader/programmer, not the computer.  Processing is cheap.  That and I carry practices from one language to another.  Lots of languages.  And a compiler with any sort of optimization turned on would inline that anyway, without any need to even request it.  Minifiers are not that smart.

Do you have a preference?
* `start + ((end - start) / 2)`
* `(start + end) / 2`

Far as I can tell, those are mathematically equivalent.  The first is what I found in the code, the second should be more efficient.  Maybe add a midpoint comment as well, to compensate for the semantics.